packages/sanity/src/structure/panes/documentList/sheetList/DocumentSheetActions.tsx

```diff
@@ -0,0 +1,205 @@
+    [schemaType, client, items],
+  )
+
+  const handlePublish = useCallback(() => {
+    setPublishStatus('publishing')
+    toast.push({
+      closable: true,
+      duration: 60000,
+      id: 'publishing-toast',
+      status: 'info',
+      title: 'Publishing',
+      description: 'Publishing documents...',
+    })
+    action
+      .execute()
+      .pipe(
+        catchError((err) => {
+          setPublishStatus('idle')
+          toast.push({
+            closable: true,
+            id: 'publishing-toast',
+            status: 'error',
+            title: 'An error occurred',
+            description: err.message,
+          })
+          return of(null)
+        }),
+      )
+      .subscribe((response) => {
+        if (response) {
+          setPublishStatus('idle')
+          toast.push({
+            closable: true,
+            id: 'publishing-toast',
+            status: 'success',
+            title: 'Published',
+            description: 'The documents have been published',
+          })
+        }
+      })
+  }, [action, toast])
+  const actionDisabled = action.disabled()
+
+  const disabled =
+    publishStatus !== 'idle' || Boolean(actionDisabled) || hasErrorStatus || isValidating
+
+  return (
+    <Box>
+      <Button
+        tone="primary"
+        tooltipProps={{
+          disabled: !disabled,
+          content: hasErrorStatus ? (
+            <Text>Cannot publish documents with validation errors</Text>
+          ) : (
+            actionDisabled && (
+              <Stack space={2}>
+                {actionDisabled.map((reason) => (
+                  <Text key={reason.id}>
+                    Item with ID {reason.id} cannot be published: <strong>{reason.reason}</strong>
+                  </Text>
+                ))}
+              </Stack>
+            )
+          ),
+        }}
+        loading={publishStatus === 'publishing'}
+        icon={PublishIcon}
+        text={'Publish all'}
+        onClick={handlePublish}
+        disabled={disabled}
+      />
+    </Box>
+  )
+}
+
+const ActionsRoot = styled(Card)(() => {
+  return css`
+    background-color: ${blue[700].hex};
+    --card-fg-color: ${white.hex};
+  `
+})
+
+const CloseButton = styled(Button)`
+  background: transparent;
+  :hover {
+    background: ${gray[200].hex};
+  }
+  --card-fg-color: ${white.hex};
+  --card-icon-color: ${white.hex};
+`
+
+const Divider = styled(Box)(() => {
+  return css`
+    width: 100%;
+    background: rgba(255, 255, 255, 0.1);
+    height: 1px;
+  `
+})
+
+const PopoverPlaceholder = styled.div`
+  position: sticky;
+  bottom: 0;
+`
+
+export function DocumentSheetActions({table, schemaType, parentRef}: DocumentSheetActionsProps) {
+  const selectedRows = table.getSelectedRowModel().rows
+  const items = useMemo(() => selectedRows.map((row) => row.original.__metadata), [selectedRows])
+  const validationStatus = useValidationStatusList(
+    items.map((item) => item.idPair.publishedId),
```

Comment:
COULD: memoize this list to avoid new reference on every render

Reply:
Yes, good catch.

Reply:
Yes good catch, thanks